locksmith/src/websub/helpers/renewFiatKey.ts

```diff
@@ -0,0 +1,200 @@
+import { Charge, KeyRenewal, KeySubscription } from '../../models'
+import Stripe from 'stripe'
+import config from '../../../config/config'
+import { Op } from 'sequelize'
+import { Web3Service } from '@unlock-protocol/unlock-js'
+import networks from '@unlock-protocol/networks'
+
+interface RenewKeyReturned {
+  keyId?: number
+  lockAddress: string
+  network: number
+  tx?: string
+  error?: string
+}
+
+interface Options {
+  lockAddress: string
+  userAddress: string
+  keyId: number
+  network: number
+}
+
+export async function renewFiatKey({
+  lockAddress,
+  userAddress,
+  keyId,
+  network,
+}: Options): Promise<RenewKeyReturned> {
+  try {
+    const renewalInfo = {
+      keyId,
+      network,
+      lockAddress,
+    }
+
+    const stripe = new Stripe(config.stripeSecret!, {
+      apiVersion: '2020-08-27',
+    })
+
+    const stripeAccount = await getStripeConnectForLock(lockAddress, network)
+
+    if (stripeAccount === 0 || stripeAccount === -1) {
+      throw new Error('No stripe connect account associated with the lock')
+    }
+
+    const subscription = await KeySubscription.findOne({
+      where: {
+        keyId,
+        lockAddress,
+        userAddress,
+        recurring: {
+          [Op.gt]: 0,
+        },
+      },
+    })
+
+    if (!subscription) {
+      throw new Error('No subscription found')
+    }
+
+    if (subscription.userAddress !== userAddress) {
+      throw new Error('Key owner is not the subscriber')
+    }
+
+    const customer = await stripe.customers.retrieve(
+      subscription.connectedCustomer,
+      {
+        stripeAccount,
+      }
+    )
+
+    if (customer.deleted) {
+      throw new Error('Customer does not exist anymore')
+    }
+
+    const paymentMethod = await stripe.paymentMethods.list(
+      {
+        customer: customer.id,
+        type: 'card',
+      },
+      {
+        stripeAccount,
+      }
+    )
+
+    const paymentMethodId = paymentMethod.data?.[0]?.id
+
+    if (!paymentMethodId) {
+      throw new Error('No payment method available on the customer profile.')
+    }
+
+    const web3Service = new Web3Service(networks)
+
+    // Get provider for network
+    const provider = await web3Service.providerForNetwork(subscription.network)
+
+    const fulfillmentDispatcher = new Dispatcher()
+
+    const paymentIntent = await stripe.paymentIntents.create(
+      {
+        amount: subscription.amount,
+        currency: 'USD',
+        capture_method: 'manual',
+        application_fee_amount: subscription.unlockServiceFee,
+        customer: customer.id,
+        payment_method: paymentMethodId,
+        metadata: {
+          lock: lockAddress,
+          keyId,
+          network,
+          recipients: [userAddress].join(','),
+          purchaser: userAddress,
+          renewal: subscription.recurring,
+        },
+      },
+      {
+        stripeAccount,
+      }
+    )
+
+    const response = await new Promise((resolve, reject) => {
```

Comment:
Can we check the payment intent's status before issuing the blockchain transaction?
https://stripe.com/docs/api/payment_intents/object#payment_intent_object-status
If the status is _not_ `require_capture` then we should _not_ send the transaction because we won't be able to capture it below. wdyt?

Reply:
Added the check.